crates/ra_assists/src/handlers/add_function.rs

```diff
@@ -224,12 +232,29 @@ fn fn_arg_name(fn_arg: &ast::Expr) -> Option<String> {
     }
 }
 
-fn fn_arg_type(ctx: &AssistCtx, fn_arg: &ast::Expr) -> Option<String> {
+fn fn_arg_type(
+    ctx: &AssistCtx,
+    target_module: &Option<hir::InFile<hir::ModuleSource>>,
+    fn_arg: &ast::Expr,
+) -> Option<String> {
     let ty = ctx.sema.type_of_expr(fn_arg)?;
     if ty.is_unknown() {
         return None;
     }
-    Some(ty.display(ctx.sema.db).to_string())
+    let target_syntax = match target_module {
+        None => fn_arg.syntax(),
```

Comment:
Hmm, taking the syntax of the function call argument as source scope might be an issue if there are use statements inside the current function... This also deserves a test

Reply:
Oh no, it doesn't work because of #1165.
I'm gonna create a test case, ignore it and refer that issue

Reply:
Well, you're taking the `module` from the scope anyway. (Also, any problems with local uses won't show up right now since we don't support them yet.)